---
title: "R Notebook"
output:
  html_document: default
  html_notebook: default
  pdf_document: default
---


## Loading and pre-processing data

```{r}

#Loading and pre-processing data

#setwd('E:/SageDocs/PredictingDriverGenes/LineageMisc/')
source('MiscPreprocessing.R')



Dat <- read.delim('Data/ROSMAP_DLPFC_logCPM.tsv',stringsAsFactors = F)
Dat2 <- read.delim('Data/ROSMAP_DLPFC_Covariates.tsv',stringsAsFactors = F)
Dat2 <- read.csv('Data/ROSMAP_DLPFC_Covariates3.csv',stringsAsFactors = F)


```



```{r}
#AMP_mods <-  read.csv('TCX_AMPAD_Modules.csv')
#AMP_mods <-  read.csv('TCX_DE.csv')
AMP_mods <-  read.csv('Data/DLPFC_DE.csv')
#In <- which(AMP_mods$LogPV.DLPFC >= 1)
#AMP_mods <- AMP_mods[In,]




#Normalize all columns 

GeneNames <- Dat$ensembl_gene_id
GeneNamesAD <- AMP_mods$GeneID

Names <- colnames(Dat)

for (i in 1:length(Names)){
  
  Names[i] <- substring(Names[i],2)
  
}


colnames(Dat) <- Names
cNames <- Dat2$SampleID
l <- length(Names)

#deleting columns not in the covariate list
temp <- rep(T,l)
for (i in 1:l){
  if (!(Names[i] %in% cNames)){
    temp[i] <- F
  }
}

In <- which(temp)
#print(temp)
Dat <- Dat[,In]

#deleting extra rows in covariate list
Names <- Names[In]
l <- length(cNames)
temp <- rep(T,l)
for (i in 1:l){
  if (!(cNames[i] %in% Names)){
    temp[i] <- F
  }
}
In <- which(temp)
Dat2 <- Dat2[In,]
  
  
DatNorm <- ColNorm(Dat)
#In_genes <- which(GeneNames %in% GeneNamesAD)
In_genes <- Get.High.Var.Genes(Dat)
DatNorm2 <- DatNorm[In_genes,]
#GeneNamesAD <- GeneNames[In_genes]

DatNorm2 <- DatNorm2[,Dat2$Batch<7]
Dat2 <- Dat2[Dat2$Batch<7,] 

library(Rtsne)
Temp <- Rtsne(t(DatNorm2))

```



## Visualizing the whole dataset using tSNE

```{r}

#tSNE visualization for whole dataset
plot(Temp$Y[,1],Temp$Y[,2], col = as.factor(Dat2$Sex))

```


## Visualizing the one brain region using tSNE


```{r}

#Keeping only TCX data 
#In_BR <- grep('TCX',Dat2$Tissue.Diagnosis)
#In_BR <- grep('DLPFC',Dat2$Tissue.Diagnosis)
#DatNorm3 <- DatNorm2[,In_BR]
#Dat3 <- Dat2[In_BR,]
DatNorm3 <- DatNorm2
Dat3 <- Dat2

Temp <- Rtsne(t(DatNorm3))
plot(Temp$Y[,1],Temp$Y[,2], col = as.factor(Dat3$Sex))

```


## Visualizing one gender using tSNE


```{r}


#Keeping only female data 
Sex <- 'FEMALE'
In_S <- which(Dat3$Sex == Sex)
DatNorm4 <- DatNorm3[,In_S]
Dat4 <- Dat3[In_S,]

In_cov <- which(Cov$ID %in% Dat4$Donor_ID)
Cov <- Cov[In_cov,]
In_cov <- c()
for(i in 1:length(Dat4$Donor_ID)){
  temp <- which(Cov$ID == Dat4$Donor_ID[i])
  In_cov <- c(In_cov,temp[1])
}
Cov <- Cov[In_cov,]

for (i in 23:26){
  Cov[,i] <- (Cov[,i] - min(Cov[,i]))/(max(Cov[,i])-min(Cov[,i]))
}

Temp <- Rtsne(t(DatNorm4))
#plot(Temp$Y[,1],Temp$Y[,2], col = as.factor(Dat4$Tissue.Diagnosis))
plot(Temp$Y[,1],Temp$Y[,2], col = as.factor(Dat4$Diagnosis.Sex))

```


## Performing lineage inference using Monocle2


```{r}

#Performing lineage inference with Monocle2
source('LineageFunctions.R')
temp <- DatNorm4
#temp2 <- cbind(Dat4,Cov)
temp2 <- Dat4
temp2$APOE4 <- as.character(temp2$APOE4)
temp2$braaksc <- as.character(temp2$braaksc)
temp2$ceradsc <- as.character(temp2$ceradsc)
temp2$cogdx.1 <- as.character(temp2$cogdx.1)

rownames(temp) <- NULL
colnames(temp) <- NULL
MonRun <- RunMonocleTobit(temp, temp2, C_by = 'endoScore')


```

## Visualizing using Monocle's visualization 

```{r}

#plot_cell_trajectory(MonRun, color_by = "Tissue.APOE4")
plot_cell_trajectory(MonRun, color_by = "Diagnosis")

```

## Performing differential expression analysis on 'resistant' samples 

```{r}
library(DESeq)
source('E:/SageDocs/CellTypeDeconv/DESeq_Helper.R')
RawDat <- read.csv('Mayo_TCX/Mayo_TCX_all_counts_matrix.csv', stringsAsFactors = F)
SortedPS <- sort(MonRun@phenoData@data$Pseudotime, index.return = TRUE)
In_control <- SortedPS$ix[which(temp2$Tissue.Diagnosis != 'TCX.AD')]
n <- floor(length(In_control)/10)
ResIn <- In_control[1:n]
NonResIn <- tail(In_control, n)
Lab <- c(rep(1,n),rep(2,n))
#Convert to counts
l <- ProcessRawDat(RawDat, Dat4, GeneNames)
RawDat <- l$RawDat
GeneNames2 <- l$GeneNames
ResDat <- RawDat[ResIn]
NonResDat <- RawDat[NonResIn]
DatTot <- cbind(ResDat, NonResDat)
rownames(DatTot) <- NULL
colnames(DatTot) <- NULL
#DatCounts <- data.frame(Convert2Counts(DatTot, Lab))
#DatCounts[DatCounts<0] <- 0

#Estimating parameters

cds = newCountDataSet( DatCounts, as.factor(as.character(Lab)) )
cds@phenoData@data$sizeFactor <- rep(1,length(Lab))
cds = estimateDispersions( cds )

#performing differential expression analysis 
res <- nbinomTest( cds, "1", "2" )
res <- res[order(res$pval),]
res$Genes <- GeneNames2[as.numeric(res$id)]

source('convertEnsemblToHgnc.R')
GeneConv <- convertEnsemblToHgnc(res$Genes)

Symb <- as.character(c(1:length(res$Genes)))

for (i in 1:length(res$Genes)){
  In <- which(GeneConv$ensembl_gene_id == res$Genes[i])
  if (length(In)>0){
    Symb[i] <- GeneConv$external_gene_name[In]
  }
}

res$Symb <- Symb

head(res)

```

## Identify genes which vary as a function of pseudotime 

```{r}
library(R.matlab)
library(utilityFunctions)
ClusterList <- 'TCX_F_pv1_k15.mat'
RM <- readMat(ClusterList)

CorrMat <- rep(0,length(GeneNamesAD))

for( i in 1:length(GeneNamesAD)){
  
  CorrMat[i] <- abs(cor(x = RM$PS, y = RM$ScMat[i,], method = 'pearson'))[1,1]
  
}

CorrDF <- as.data.frame(CorrMat)

GeneConv2 <- convertEnsemblToHgnc(GeneNamesAD)

Symb2 <- as.character(c(1:length(GeneNamesAD)))

for (i in 1:length(GeneNamesAD)){
  In <- which(GeneConv2$ensembl_gene_id == GeneNamesAD[i])
  if (length(In)>0){
    Symb2[i] <- GeneConv2$external_gene_name[In]
  }
}

CorrDF$ENSG <- GeneNamesAD
CorrDF$Symb <- Symb2

head(CorrDF)


```


## Perform enrichment analysis using IGAP GWAS

```{r}

IGAP_list <- read.csv('IGAP_gene_summary.csv')
CutOff <- 0.7
PS_genes <- CorrDF$Symb[which(CorrMat>CutOff)]

'%ni%' <- Negate('%in%')

In_igap1 <- which(IGAP_list$Names %in% PS_genes)
In_igap2 <- which(IGAP_list$Names %ni% PS_genes)

print(t.test(IGAP_list$Mean[In_igap1],IGAP_list$Mean[In_igap2]))
print(t.test(log10(IGAP_list$Min[In_igap1]),log10(IGAP_list$Min[In_igap2])))



```

