---
title: "All Paper Figures"
output: html_notebook
---

Recreate the original analysis with files in synapse (Ben's original code). Create boxplot and summary statistics of AD vs pseudo-time association in TCX and DLPFC.  First pull relevant data and do join.

```{r}
synapser::synLogin()
tcxLineageTimes <- synapser::synTableQuery("select * from syn17023721")$asDataFrame()
dlpfcLineageTimes <- synapser::synTableQuery("select * from syn17023795")$asDataFrame()
tcxCovObj <- synapser::synGet("syn8466814")
tcxCov <- data.table::fread(tcxCovObj$path,data.table=F)
dlpfcCovObj <- synapser::synGet("syn11024258")
dlpfcCov <- data.table::fread(dlpfcCovObj$path,data.table=F)
tcxLineageTimes <- tcxLineageTimes[,-c(1:3)]
dlpfcLineageTimes <- dlpfcLineageTimes[,-c(1:3)]
dlpfc <- dplyr::left_join(dlpfcLineageTimes,dlpfcCov,by='SampleID')
tcx <- dplyr::left_join(tcxLineageTimes,tcxCov,by='SampleID')
```

Next run a logistic regression

```{r}
tcxAD <- rep(NA,nrow(tcx))
tcxAD[tcx$Tissue.Diagnosis == 'TCX.AD'] <- 1
tcxAD[tcx$Tissue.Diagnosis == 'TCX.CONTROL'] <- 0
tcxDf <- data.frame(diagnosis=tcxAD,
                    pseudotime=tcx$Pseudotime,
                    stringsAsFactors = FALSE)
tcxDf$pseudotime <- scale(tcxDf$pseudotime,center=F)
summary(glm(diagnosis ~ pseudotime,tcxDf,family='binomial'))

dlpfcAD <- rep(NA,nrow(dlpfc))
dlpfcAD[dlpfc$Diagnosis == 'AD'] <- 1
dlpfcAD[dlpfc$Diagnosis == 'CONTROL'] <- 0
dlpfcDf <- data.frame(diagnosis = dlpfcAD,
                      pseudotime=dlpfc$Pseudotime,
                      stringsAsFactors = FALSE)
dlpfcDf$pseudotime <- scale(dlpfcDf$pseudotime,center=F)
summary(glm(diagnosis ~ pseudotime,dlpfcDf,family='binomial'))

```

Next make a ggplot boxplot and save to figures folder.  First combine into a single df

```{r}
tcxDf$BrainRegion <- 'TCX'
dlpfcDf$BrainRegion <- 'DLPFC'
combinedDf <- rbind(tcxDf,dlpfcDf)
combinedDf <- combinedDf[!is.na(combinedDf$diagnosis),]
combinedDf$diagnosis[combinedDf$diagnosis==1] <- 'AD'
combinedDf$diagnosis[combinedDf$diagnosis==0] <- 'Control'
combinedDf$diagnosis <- factor(combinedDf$diagnosis,levels=c('AD','Control'))
combinedDf$BrainRegion <- as.factor(combinedDf$BrainRegion)

```

Next make ggplot

```{r}
g <- ggplot2::ggplot(combinedDf,ggplot2::aes(x=BrainRegion,
                                             y=pseudotime,
                                             color=diagnosis))
g <- g + ggplot2::geom_boxplot()
g <- g + ggplot2::geom_point(position=ggplot2::position_jitterdodge())
#g <- g + ggplot2::scale_color_viridis_d()
g <- g + ggplot2::scale_color_manual(values=viridis::viridis(3)[1:2])
tiff(file='~/AMPAD_Lineage/LH_reviewer_response/Original_boxplots_LH.tiff',height=85,width=100,units='mm',res=300)
g
dev.off()
ggplot2::ggsave('ADcasecontrol_boxplots_original.tiff')
```
Repeat the regression and boxplots with the pseudotimes adjusted for: RIN number, PMI, principal components for genetic ancestry, and all three factors combined
```{r}
tcxLineageTimesLH <- read.csv("~/AMPAD_Lineage/LH_reviewer_response/TCX_F_PStimes_adjusted.csv", stringsAsFactors = FALSE)
dlpfcLineageTimesLH <- read.csv("~/AMPAD_Lineage/LH_reviewer_response/DLPFC_F_PStimes_adjusted.csv", stringsAsFactors = FALSE)
tcxCovObj <- synapser::synGet("syn8466814")
tcxCov <- data.table::fread(tcxCovObj$path,data.table=F)
dlpfcCovObj <- synapser::synGet("syn11024258")
dlpfcCov <- data.table::fread(dlpfcCovObj$path,data.table=F)
tcxLineageTimesLH <- tcxLineageTimesLH[,-c(1)]
dlpfcLineageTimesLH <- dlpfcLineageTimesLH[,-c(1)]
dlpfc <- dplyr::left_join(dlpfcLineageTimesLH,dlpfcCov,by='SampleID')
tcx <- dplyr::left_join(tcxLineageTimesLH,tcxCov,by='SampleID')
```
scale and compare pseudotimes

```{r}
tcx$pseudotime_unadj_sc <- scale(tcx$Pseudotime_unadj,center=F)
tcx$pseudotime_rin_sc <- scale(tcx$Pseudotime_RIN,center=F)
tcx$pseudotime_pmi_sc <- scale(tcx$Pseudotime_PMI,center=F)
tcx$pseudotime_pcs_sc <- scale(tcx$Pseudotime_PCs,center=F)
tcx$pseudotime_all_sc <- scale(tcx$Pseudotime_ALL,center=F)

dlpfc$pseudotime_unadj_sc <- scale(dlpfc$Pseudotime_unadj,center=F)
dlpfc$pseudotime_rin_sc <- scale(dlpfc$Pseudotime_RIN,center=F)
dlpfc$pseudotime_pmi_sc <- scale(dlpfc$Pseudotime_PMI,center=F)
dlpfc$pseudotime_pcs_sc <- scale(dlpfc$Pseudotime_PCs,center=F)
dlpfc$pseudotime_all_sc <- scale(dlpfc$Pseudotime_ALL,center=F)

t.test(tcx$pseudotime_unadj_sc, tcx$pseudotime_rin_sc)
t.test(tcx$pseudotime_unadj_sc, tcx$pseudotime_pmi_sc)
t.test(tcx$pseudotime_unadj_sc, tcx$pseudotime_pcs_sc)
t.test(tcx$pseudotime_unadj_sc, tcx$pseudotime_all_sc)

t.test(dlpfc$pseudotime_unadj_sc, dlpfc$pseudotime_rin_sc)
t.test(dlpfc$pseudotime_unadj_sc, dlpfc$pseudotime_pmi_sc)
t.test(dlpfc$pseudotime_unadj_sc, dlpfc$pseudotime_pcs_sc)
t.test(dlpfc$pseudotime_unadj_sc, dlpfc$pseudotime_all_sc)

```
Next run a logistic regression

```{r}
tcxAD <- rep(NA,nrow(tcx))
tcxAD[tcx$Tissue.Diagnosis == 'TCX.AD'] <- 1
tcxAD[tcx$Tissue.Diagnosis == 'TCX.CONTROL'] <- 0
tcxDf <- data.frame(diagnosis=tcxAD,
                    pseudotime_rin=tcx$Pseudotime_RIN,
                    pseudotime_pmi=tcx$Pseudotime_PMI,
                    pseudotime_pcs=tcx$Pseudotime_PCs,
                    pseudotime_all=tcx$Pseudotime_ALL,
                    stringsAsFactors = FALSE)
tcxDf$pseudotime_rin <- scale(tcxDf$pseudotime_rin,center=F)
tcxDf$pseudotime_pmi <- scale(tcxDf$pseudotime_pmi,center=F)
tcxDf$pseudotime_pcs <- scale(tcxDf$pseudotime_pcs,center=F)
tcxDf$pseudotime_all <- scale(tcxDf$pseudotime_all,center=F)
summary(glm(diagnosis ~ pseudotime_rin,tcxDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_pmi,tcxDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_pcs,tcxDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_all,tcxDf,family='binomial'))

dlpfcAD <- rep(NA,nrow(dlpfc))
dlpfcAD[dlpfc$Diagnosis == 'AD'] <- 1
dlpfcAD[dlpfc$Diagnosis == 'CONTROL'] <- 0
dlpfcDf <- data.frame(diagnosis = dlpfcAD,
                      pseudotime_rin=dlpfc$Pseudotime_RIN,
                      pseudotime_pmi=dlpfc$Pseudotime_PMI,
                      pseudotime_pcs=dlpfc$Pseudotime_PCs,
                      pseudotime_all=dlpfc$Pseudotime_ALL,
                      stringsAsFactors = FALSE)
dlpfcDf$pseudotime_rin <- scale(dlpfcDf$pseudotime_rin,center=F)
dlpfcDf$pseudotime_pmi <- scale(dlpfcDf$pseudotime_pmi,center=F)
dlpfcDf$pseudotime_pcs <- scale(dlpfcDf$pseudotime_pcs,center=F)
dlpfcDf$pseudotime_all <- scale(dlpfcDf$pseudotime_all,center=F)
summary(glm(diagnosis ~ pseudotime_rin,dlpfcDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_pmi,dlpfcDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_pcs,dlpfcDf,family='binomial'))
summary(glm(diagnosis ~ pseudotime_all,dlpfcDf,family='binomial'))

```

Next make a ggplot boxplot and save.  First combine into a single df
```{r}
tcxDf$BrainRegion <- 'TCX'
dlpfcDf$BrainRegion <- 'DLPFC'
combinedDf <- rbind(tcxDf,dlpfcDf)
combinedDf <- combinedDf[!is.na(combinedDf$diagnosis),]
combinedDf$diagnosis[combinedDf$diagnosis==1] <- 'AD'
combinedDf$diagnosis[combinedDf$diagnosis==0] <- 'Control'
combinedDf$diagnosis <- factor(combinedDf$diagnosis,levels=c('AD','Control'))
combinedDf$BrainRegion <- as.factor(combinedDf$BrainRegion)
```
run ggplot2 for rin-adjusted pseudotimes
```{r}
g <- ggplot2::ggplot(combinedDf,ggplot2::aes(x=BrainRegion,
                                             y=pseudotime_rin,
                                             color=diagnosis))
g <- g + ggplot2::geom_boxplot() + theme(text = element_text(size = 22)) 
g <- g + ggplot2::geom_point(size=2.5, position=ggplot2::position_jitterdodge())
#g <- g + ggplot2::scale_color_viridis_d()
g <- g + ggplot2::scale_color_manual(values=viridis::viridis(3)[1:2])
#tiff(file='~/AMPAD_Lineage/LH_reviewer_response/RecreateMO_boxplots_LH.tiff',height=85,width=100,units='mm',res=300)
g
#dev.off()
ggplot2::ggsave('AD_PSrin_adj_boxplots_LH.tiff')
```
run ggplot2 for pmi-adjusted pseudotimes
```{r}
g <- ggplot2::ggplot(combinedDf,ggplot2::aes(x=BrainRegion,
                                             y=pseudotime_pmi,
                                             color=diagnosis))
g <- g + ggplot2::geom_boxplot() + theme(text = element_text(size = 22)) 
g <- g + ggplot2::geom_point(size=2.5, position=ggplot2::position_jitterdodge())
#g <- g + ggplot2::scale_color_viridis_d()
g <- g + ggplot2::scale_color_manual(values=viridis::viridis(3)[1:2])
#tiff(file='~/AMPAD_Lineage/LH_reviewer_response/RecreateMO_boxplots_LH.tiff',height=85,width=100,units='mm',res=300)
g
#dev.off()
ggplot2::ggsave('AD_PSpmi_adj_boxplots_LH.tiff')
```
run ggplot2 for genetic ancestry prinicipal components-adjusted pseudotimes
```{r}
g <- ggplot2::ggplot(combinedDf,ggplot2::aes(x=BrainRegion,
                                             y=pseudotime_pcs,
                                             color=diagnosis))
g <- g + ggplot2::geom_boxplot() + theme(text = element_text(size = 22)) 
g <- g + ggplot2::geom_point(size=2.5, position=ggplot2::position_jitterdodge())
#g <- g + ggplot2::scale_color_viridis_d()
g <- g + ggplot2::scale_color_manual(values=viridis::viridis(3)[1:2])
#tiff(file='~/AMPAD_Lineage/LH_reviewer_response/RecreateMO_boxplots_LH.tiff',height=85,width=100,units='mm',res=300)
g
#dev.off()
ggplot2::ggsave('AD_PSpc_adj_boxplots_LH.tiff')
```

run ggplot2 for pseudotimes adjusted for rin, pmi, and ancestry principal components
```{r}
g <- ggplot2::ggplot(combinedDf,ggplot2::aes(x=BrainRegion,
                                             y=pseudotime_all,
                                             color=diagnosis))
g <- g + ggplot2::geom_boxplot() + theme(text = element_text(size = 22)) 
g <- g + ggplot2::geom_point(size=2.5, position=ggplot2::position_jitterdodge())
#g <- g + ggplot2::scale_color_viridis_d()
g <- g + ggplot2::scale_color_manual(values=viridis::viridis(3)[1:2])
#tiff(file='~/AMPAD_Lineage/LH_reviewer_response/RecreateMO_boxplots_LH.tiff',height=85,width=100,units='mm',res=300)
g
#dev.off()
ggplot2::ggsave('AD_PSall_adj_boxplots_LH.tiff')
```
```



Next do neuropathology associations.  First pull ROSMAP covariate file.
```{r}
rosmapObj <- synapser::synGet('syn3191087')
rosmap <- data.table::fread(rosmapObj$path,data.table=F)
rosmapIdObj <- synapser::synGet('syn3382527')
rosmapId <- data.table::fread(rosmapIdObj$path,data.table=F)
rosmapId <- dplyr::select(rosmapId,projid,rnaseq_id)
rosmapRNAid<-dplyr::left_join(rosmapId,rosmap)
dlpfcComplete<-dplyr::left_join(dlpfcLineageTimes,rosmapRNAid,by=c('SampleID'='rnaseq_id'))
dlpfcComplete$Pseudotime <- scale(dlpfcComplete$Pseudotime,center=F)

dlpfcComplete <- dplyr::select(dlpfcComplete,Pseudotime,braaksc,ceradsc,cogdx)
dlpfcComplete$braaksc <- factor(dlpfcComplete$braaksc,levels = c(0:6))
dlpfcComplete$ceradsc <- factor(dlpfcComplete$ceradsc,levels = c(1:4))
dlpfcComplete$cogdxNew <- rep(NA,nrow(dlpfcComplete))
dlpfcComplete$cogdxNew[dlpfcComplete$cogdx==1] <- 'NCI'
dlpfcComplete$cogdxNew[dlpfcComplete$cogdx==2] <- 'MCI'
dlpfcComplete$cogdxNew[dlpfcComplete$cogdx==4] <- 'LOAD'
dlpfcComplete$cogdxNew <- factor(dlpfcComplete$cogdxNew,levels = c('NCI','MCI','LOAD'))

dlpfcComplete <- dlpfcComplete[!duplicated(dlpfcComplete),]

braakfit <- MASS::polr(braaksc ~ Pseudotime,dlpfcComplete)
ceradfit <- MASS::polr(ceradsc ~ Pseudotime,dlpfcComplete)
cogdxfit <- MASS::polr(cogdxNew ~ Pseudotime,dlpfcComplete)

cat('braak p-value: ',pt(abs(summary(braakfit)$coef[1,3]),braakfit$df.residual,lower.tail=F)*2,'\n')
cat('cerad p-value: ',pt(abs(summary(ceradfit)$coef[1,3]),ceradfit$df.residual,lower.tail=F)*2,'\n')
cat('cogdx p-value: ',pt(abs(summary(cogdxfit)$coef[1,3]),cogdxfit$df.residual,lower.tail=F)*2,'\n')
```
GWAS LOAD distributions
(needed to tweak the function to change the host site)
```{r}

convertEnsemblToHgnc <- function(ensemblIds){
  
  ensembl=biomaRt::useMart('ENSEMBL_MART_ENSEMBL',
                           dataset = 'hsapiens_gene_ensembl',
                           host='useast.ensembl.org')
  
  genes<-getBM(attributes = c('ensembl_gene_id','external_gene_name'),
               filters='ensembl_gene_id',
               values=ensemblIds,
               mart=ensembl)
  return(genes)
}
Make.Gene.Symb <- function(GeneENSG){
  
  #source('convertEnsemblToHgnc.R')
  GeneConv <- convertEnsemblToHgnc(GeneENSG)
  Symb <- as.character(c(1:length(GeneENSG)))
  
  for (i in 1:length(GeneENSG)){
    In <- which(GeneConv$ensembl_gene_id == GeneENSG[i])
    if (length(In)>0){
      Symb[i] <- GeneConv$external_gene_name[In]
    }
  }
  
  return(Symb)
}

```
add expression data to tcx data frame, calculate correlations
```{r}

tcxCPMObj <- synapser::synGet('syn8466816')
Dat <- data.table::fread(tcxCPMObj$path,data.table=F)
sampleIds <- colnames(Dat)[-1]
geneIds <- Dat$ensembl_gene_id
Dat <- Dat[,-1]
Dat <- t(Dat)
colnames(Dat) <- geneIds
Dat <- data.frame(Dat,stringsAsFactors=F)
Dat$sampleId <- sampleIds
tcx <- dplyr::left_join(tcx,Dat,by=c('SampleID'='sampleId'))
tcx2 <- dplyr::select(tcx,dplyr::starts_with("ENSG"))
corvec <- cor(tcx2,tcx$Pseudotime,method='spearman')
corDf <- data.frame(geneid=colnames(tcx2),cor=corvec,stringsAsFactors=F)
corDf2 <- corDf
corDf2$external_gene_name <- Make.Gene.Symb(corDf2$geneid)
corDf2$cor <- NULL
corDf2 <- dplyr::left_join(corDf,corDf2,by=c('geneid'))

```
and add expression data to dlpfc data frame, calculate correlations
```{r}
dlpfcCPMObj <- synapser::synGet('syn8456638')
Dat <- data.table::fread(dlpfcCPMObj$path,data.table=F)
sampleIds <- colnames(Dat)[-1]
geneIds <- Dat$ensembl_gene_id
Dat <- Dat[,-1]
Dat <- t(Dat)
colnames(Dat) <- geneIds
Dat <- data.frame(Dat,stringsAsFactors=F)
Dat$sampleId <- sampleIds
dlpfc <- dplyr::left_join(dlpfc,Dat,by=c('SampleID'='sampleId'))
dlpfc2 <- dplyr::select(dlpfc,dplyr::starts_with("ENSG"))
corvec <- cor(dlpfc2,dlpfc$Pseudotime,method='spearman')
corDfdlpfc <- data.frame(geneid=colnames(dlpfc2),cor=corvec,stringsAsFactors=F)
corDfdlpfc2 <- corDfdlpfc
corDfdlpfc2$external_gene_name <- Make.Gene.Symb(corDfdlpfc2$geneid)
corDfdlpfc2$cor <- NULL
corDfdlpfc2 <- dplyr::left_join(corDfdlpfc,corDfdlpfc2,by=c('geneid'))

```
input desired gwas genes
```{r}
ad_gwas <- c("CR1",
             "BIN1",
             "INPP5D",
             "HLA-DRB1",
             "TREM2",
             "MEF2C",
             "NME8",
             "CD2AP",
             "NYAP1",
             "EPHA1",
             "PTK2B",
             "CLU",
             "SPI1",
             "MS4A2",
             "PICALM",
             "SORL1",
             "FERMT2",
             "SLC24A4",
             "ABCA7",
             "APOE",
             "CASS4",
             "ECHDC3",
             "ACE",
             "NDUFAF6",
             "ECHDC3",
             "ADAMTS20",
             "SPPL2A",
             "ADAM10",
             "IQCK",
             "MIR142",
             "ACE",
             "ADAMTS1",
             "SUCLG2P4",
             "FST",
             "OARD1",
             "WWOX",
             "MAF",
             "CD55",
             "YOD1",
             "HLA-DRB1",
             "PSMB8",
             "C4A",
             "GPSM3",
             "HLA-DPA1",
             "HLA-DQA1",
             "HLA-DRA",
             "HLA-DRB5",
             "PSMB9",
             "CD2AP",
             "AGFG2",
             "PILRA",
             "EPHB4",
             "C7orf43",
             "GAL3ST4",
             "ZKSCCAN1",
             "FAM131B",
             "PSMC3",
             "ACP2",
             "C1QTNF4",
             "CELF1",
             "MTCH2",
             "NDUFS3",
             "NUP160",
             "MS4A6A",
             "MS4A7",
             "MS4A4A",
             "EED",
             "PICALM",
             "STYX",
             "RIN3",
             "HMHA1",
             "CNN2",
             "WDR18",
             "CASS4")
```

```{r}
corDf <- corDf2
mean(abs(corDf$cor))

mean(abs(corDf[corDf$external_gene_name %in% ad_gwas,]$cor))


corDf$adGwas <- corDf$external_gene_name %in% ad_gwas
corDfdlpfc$adGwas <- corDfdlpfc$external_gene_name %in% ad_gwas
corDf$brainRegion <- 'TCX'
corDfdlpfc$brainRegion <- 'DLPFC'
corDfcombined <- rbind(corDf,corDfdlpfc)
colnames(corDfcombined)[4] <- 'LOADGWASGene'
g <- ggplot2::ggplot(corDfcombined,ggplot2::aes(x=brainRegion,y=cor,fill=LOADGWASGene))
g <- g + ggplot2::geom_boxplot()
g <- g + ggplot2::scale_fill_viridis_d()
g <- g + ggplot2::geom_point(position=ggplot2::position_jitterdodge())
g <- g + ggplot2::labs(x = 'Brain Region',y='Correlation with pseudotime',fill='LOAD\nGWAS\nGene')

#g <- g + ggplot2::geom_point(position=ggplot2::position_jitterdodge())
g
# tiff(file='~/Desktop/MANUSCRIPT/figure2d.tiff',height=85,width=100,units='mm',res=300)
g
dev.off()
```


```{r}
```


```{r}
```


```{r}
```

```

```


Making scatter plot of association with pseudotime and gwas summary statistics.  Get Kunkle statistics

```{r}
# kunkleStats <- data.table::fread('Kunkle_etal_Stage1_results.txt',data.table=F)
# kunkleStatsTest <- kunkleStats
# colnames(kunkleStatsTest) <- c('#CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO')
# kunkleStatsTest$QUAL <- 30
# kunkleStatsTest$FILTER <- 'PASS'
# kunkleStatsTest$INFO <- ''
# write.table(kunkleStatsTest,file='kunkle.vcf',sep = '\t',row.names=F,quote=F)
```

run snpEff to annotate variants appropriately (with annotate_variants.sh)

```{r}
kunkleAnnotated <- data.table::fread('kunkle.final.ann.vcf',data.table=F,skip=5)
```



write function to split rows into 


```{r}
buildRecords <- function(chrom,pos,id,ref,alt,info){
  foo <- strsplit(info,',')
  bar <- lapply(foo,strsplit,'\\|')
  bar <- unlist(bar,recursive=FALSE)
  bar <- sapply(bar,function(x) x[4])
  bar <- unique(bar)
  df <- data.frame(chrom=chrom,
                   pos=pos,
                   id=id,
                   ref=ref,
                   alt=alt,
                   GeneID=bar,
                   stringsAsFactors=F)
  return(df)
}

buildRecords2 <- function(id,info){
  foo <- strsplit(info,',')
  bar <- lapply(foo,strsplit,'\\|')
  bar <- unlist(bar,recursive=FALSE)
  bar <- sapply(bar,function(x) x[4])
  bar <- unique(bar)
  df <- data.frame(id=id,
                   GeneID=bar,
                   stringsAsFactors=F)
  return(df)
}

system.time(res <- mapply(buildRecords,
              kunkleAnnotated$`#CHROM`[1:1000],
              kunkleAnnotated$POS[1:1000],
              kunkleAnnotated$ID[1:1000],
              kunkleAnnotated$REF[1:1000],
              kunkleAnnotated$ALT[1:1000],
              kunkleAnnotated$INFO[1:1000],
              SIMPLIFY=FALSE))

system.time(res <- mapply(buildRecords2,
              kunkleAnnotated$ID,
              kunkleAnnotated$INFO,
              SIMPLIFY=FALSE))

res <- do.call(rbind,res)
kunkleStats <- data.table::fread('Kunkle_etal_Stage1_results.txt',data.table=F)
combinedFrame <- dplyr::left_join(res,kunkleStats,by=c('id'='MarkerName'))
write.table(combinedFrame,file='kunkle_stage1_results_gene_20kb.txt',sep='\t',row.names=F,quote=F)

```

group by and summarize

```{r}
combinedFrame$Pvalue <- as.numeric(combinedFrame$Pvalue)
foobar <- dplyr::group_by(combinedFrame,GeneID)
pval <- dplyr::summarise(foobar,meanPval=mean(-log10(Pvalue)))
```


read in count data and compute summaries.


```{r}
#synapser::synLogin()

tcxCPMObj <- synapser::synGet('syn8466816')
#Dat <- read.delim(tcxCPMObj$path,stringsAsFactors = F)
Dat <- data.table::fread(tcxCPMObj$path,data.table=F)
sampleIds <- colnames(Dat)[-1]
geneIds <- Dat$ensembl_gene_id
Dat <- Dat[,-1]
Dat <- t(Dat)
colnames(Dat) <- geneIds
Dat <- data.frame(Dat,stringsAsFactors=F)
Dat$sampleId <- sampleIds
tcx <- dplyr::left_join(tcx,Dat,by=c('SampleID'='sampleId'))
tcx2 <- dplyr::select(tcx,dplyr::starts_with("ENSG"))
corvec <- cor(tcx2,tcx$Pseudotime,method='spearman')
corDf <- data.frame(geneid=colnames(tcx2),cor=corvec,stringsAsFactors=F)
map <- utilityFunctions::convertEnsemblToHgnc(corDf$geneid)
corDf <- dplyr::left_join(corDf,map,by=c('geneid'='ensembl_gene_id'))

#corDf <- dplyr::left_join(corDf,pval,c('external_gene_name'='GeneID'))

```

dlpfc
```{r}
dlpfcCPMObj <- synapser::synGet('syn8456638')
Dat <- data.table::fread(dlpfcCPMObj$path,data.table=F)
sampleIds <- colnames(Dat)[-1]
geneIds <- Dat$ensembl_gene_id
Dat <- Dat[,-1]
Dat <- t(Dat)
colnames(Dat) <- geneIds
Dat <- data.frame(Dat,stringsAsFactors=F)
Dat$sampleId <- sampleIds
dlpfc <- dplyr::left_join(dlpfc,Dat,by=c('SampleID'='sampleId'))
dlpfc2 <- dplyr::select(dlpfc,dplyr::starts_with("ENSG"))



corvec <- cor(dlpfc2,dlpfc$Pseudotime,method='spearman')
corDfdlpfc <- data.frame(geneid=colnames(dlpfc2),cor=corvec,stringsAsFactors=F)
map <- utilityFunctions::convertEnsemblToHgnc(corDfdlpfc$geneid)
corDfdlpfc <- dplyr::left_join(corDfdlpfc,map,by=c('geneid'='ensembl_gene_id'))



```



```{r}
ad_gwas <- c("CR1",
             "BIN1",
             "INPP5D",
             "HLA-DRB1",
             "TREM2",
             "MEF2C",
             "NME8",
             "CD2AP",
             "NYAP1",
             "EPHA1",
             "PTK2B",
             "CLU",
             "SPI1",
             "MS4A2",
             "PICALM",
             "SORL1",
             "FERMT2",
             "SLC24A4",
             "ABCA7",
             "APOE",
             "CASS4",
             "ECHDC3",
             "ACE",
             "NDUFAF6",
             "ECHDC3",
             "ADAMTS20",
             "SPPL2A",
             "ADAM10",
             "IQCK",
             "MIR142",
             "ACE",
             "ADAMTS1",
             "SUCLG2P4",
             "FST",
             "OARD1",
             "WWOX",
             "MAF",
             "CD55",
             "YOD1",
             "HLA-DRB1",
             "PSMB8",
             "C4A",
             "GPSM3",
             "HLA-DPA1",
             "HLA-DQA1",
             "HLA-DRA",
             "HLA-DRB5",
             "PSMB9",
             "CD2AP",
             "AGFG2",
             "PILRA",
             "EPHB4",
             "C7orf43",
             "GAL3ST4",
             "ZKSCCAN1",
             "FAM131B",
             "PSMC3",
             "ACP2",
             "C1QTNF4",
             "CELF1",
             "MTCH2",
             "NDUFS3",
             "NUP160",
             "MS4A6A",
             "MS4A7",
             "MS4A4A",
             "EED",
             "PICALM",
             "STYX",
             "RIN3",
             "HMHA1",
             "CNN2",
             "WDR18",
             "CASS4")

mean(abs(corDf$cor))

mean(abs(corDf[corDf$external_gene_name %in% ad_gwas,]$cor))


corDf$adGwas <- corDf$external_gene_name %in% ad_gwas
corDfdlpfc$adGwas <- corDfdlpfc$external_gene_name %in% ad_gwas
corDf$brainRegion <- 'TCX'
corDfdlpfc$brainRegion <- 'DLPFC'
corDfcombined <- rbind(corDf,corDfdlpfc)
colnames(corDfcombined)[4] <- 'LOADGWASGene'
g <- ggplot2::ggplot(corDfcombined,ggplot2::aes(x=brainRegion,y=cor,fill=LOADGWASGene))
g <- g + ggplot2::geom_boxplot()
g <- g + ggplot2::scale_fill_viridis_d()
g <- g + ggplot2::geom_point(position=ggplot2::position_jitterdodge())
g <- g + ggplot2::labs(x = 'Brain Region',y='Correlation with pseudotime',fill='LOAD\nGWAS\nGene')

#g <- g + ggplot2::geom_point(position=ggplot2::position_jitterdodge())
g
# tiff(file='~/Desktop/MANUSCRIPT/figure2d.tiff',height=85,width=100,units='mm',res=300)
g
dev.off()
```
